frontend/src/features/interest_points/EditInterestPoint.js

```diff
@@ -32,82 +31,52 @@ const SaveInterestPoint = ({ healthcheckTextWarning, isOpen, close }) => {
   } = useSelector(state => state.interestPoint)
 
   /** @type {number[]} coordinates - Coordinates formatted in DD [latitude, longitude] */
-  const [coordinates, setCoordinates] = useState([])
-  const [name, setName] = useState('')
-  const [observations, setObservations] = useState('')
-  const [type, setType] = useState()
-  const [updateInterestPoint, setUpdateInterestPoint] = useState(false)
-
-  useEffect(() => {
-    if (!isOpen) {
-      setName('')
-      setObservations('')
-      setType(interestPointType.OTHER)
-      setCoordinates([])
-      return
+  const coordinates = useMemo(() => {
+    if (!interestPointBeingDrawed?.coordinates?.length) {
+      return []
     }
 
-    setUpdateInterestPoint(false)
-    setName(interestPointBeingDrawed?.name || '')
-    setObservations(interestPointBeingDrawed?.observations || '')
-    setType(interestPointBeingDrawed?.type || interestPointType.OTHER)
-  }, [interestPointBeingDrawed, isEditing, isOpen])
+    const ddCoordinates = getCoordinates(interestPointBeingDrawed.coordinates, OPENLAYERS_PROJECTION, CoordinatesFormat.DECIMAL_DEGREES, false)
 
-  useEffect(() => {
-    if (type && !interestPointBeingDrawed?.type && !updateInterestPoint) {
-      setUpdateInterestPoint(true)
-    }
-  }, [type, interestPointBeingDrawed, updateInterestPoint])
-
-  useEffect(() => {
-    if (isOpen) {
-      if (!interestPointBeingDrawed?.coordinates?.length) {
-        setCoordinates([])
-        return
-      }
-
-      const ddCoordinates = getCoordinates(interestPointBeingDrawed.coordinates, OPENLAYERS_PROJECTION, CoordinatesFormat.DECIMAL_DEGREES, false)
+    return [
+      parseFloat(ddCoordinates[0].replace(/°/g, '')),
+      parseFloat(ddCoordinates[1].replace(/°/g, ''))
+    ]
+  }, [interestPointBeingDrawed?.coordinates])
 
-      setCoordinates([
-        parseFloat(ddCoordinates[0].replace(/°/g, '')),
-        parseFloat(ddCoordinates[1].replace(/°/g, ''))
-      ])
-    }
-  }, [interestPointBeingDrawed, isEditing, isOpen])
-
-  useEffect(() => {
-    if (name && interestPointBeingDrawed?.name !== name && updateInterestPoint) {
+  const updateName = useCallback(name => {
+    if (name && interestPointBeingDrawed?.name !== name) {
       dispatch(updateInterestPointKeyBeingDrawed({
         key: 'name',
         value: name
       }))
     }
-  }, [name, interestPointBeingDrawed, updateInterestPoint])
+  }, [interestPointBeingDrawed?.name])
```

Comment:
En tant que dependance sur des objets qui peuvent etre `undefined`, autant directement watcher `interestPointBeingDrawed` (sans le `?....`).

TODO for me

Dans l'ideal je me demande s'il n'y a pas une meilleur pratique si on utilise un objet comme un state avec des props qui sont sensees etre modifiees regulierement.